r/wrds-crsp-and-compustat.qmd

```diff
@@ -343,8 +355,22 @@ for (j in 1:batches) {
   ]
 
   crsp_daily_sub <- dsf_db |>
-    filter(permno %in% permno_batch &
-             dlycaldt >= start_date & dlycaldt <= end_date) |>
+    filter(permno %in% permno_batch) |> 
+    filter(dlycaldt >= start_date & dlycaldt <= end_date) |>
+    filter(primaryexch %in% c("N", "A", "Q"),
+           conditionaltype %in% c("RW", "NW"),
+           tradingstatusflg == "A") |> 
+    inner_join(
+      stksecurityinfohist_db |>
+        filter(sharetype == "NS" & 
```

Comment:
I want to avoid having the filter() statements twice because of duplicate code to 338-342. Can we instead just do left_join(stksecurityinfohist_db |> select(permno, secinfostartdt, secinfoenddt)) because the main purpose is to get the two date columns anyway?

Reply:
I am fine with removing 338-342's filter.

For 365ff: consider a stock that changes in one of the filter attributes (creating a new permno entry with distinct dates). Then, we would keep it in the sample even though it should be deleted. Ergo, I think these filters are necessary.

I also prefer inner_join, but would consider something else.